libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/EHorizon.kt

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.navigation.base.trip.model
+
+/**
+ * The EHorizon
+ *
+ * @param start
+ */
+data class EHorizon(
+    var start: Edge
+) {
+
+    /**
+     * Get the current edge given the position
+     */
+    fun current(position: EHorizonPosition): Edge {
+        var edge: Edge? = start
+        // TODO Check EHorizonPosition included the Edge and not the edgeId in the original design
+        // while (edge != position.edge) {
+        while (edge?.id != position.edgeId) {
```

Comment:
If `out` array will contain 2 branches with 0-level(it is possible actually) it will take first one, right? In this case it may not work. But in reality at the moment your current position is always on the very first edge of e-horizon, so it will work for now(but we may change this logic in the future).